src/number_theory/modular_forms/slash_actions.lean

```diff
@@ -26,38 +26,38 @@ local notation `GL(` n `, ` R `)`⁺ := matrix.GL_pos (fin n) R
 local notation `SL(` n `, ` R `)` := matrix.special_linear_group (fin n) R
 
 /--A general version of the slash action of the space of modular forms.-/
-class slash_action (β : Type*) (G : Type*) (α : Type*) (γ : Type*) [group G] [ring α]
-  [has_smul γ α] :=
-(map : β → G → α → α)
+class slash_action (β G α δ γ : Type*) [group G] [has_zero (α → δ)]
+[has_one (α → δ)] [has_smul γ δ] [has_add (α → δ)] :=
```

Comment:
I don't understand this change; do you want to talk about `1 : α → δ` that is not defined with `1 a = 1`? If not, this should be
```suggestion
class slash_action (β G α δ γ : Type*) [group G] [has_zero δ]
[has_one δ] [has_smul γ δ] [has_add δ] :=
```
But do you need to specialize to `α → δ` in the first place? Why not just leave it as `α` and use `α → δ` at the call site?

Reply:
Probably not. I just wanted to go with what seemed the least restrictive version.

Reply:
The least restrictive version is the one in master IMO

Reply:
Oh you're right, I was being silly and  misremembering what was originally there!